### ISSUE TITLE
Add documentation for Datasource constructor

### ISSUE THREAD

**Issue by davidtheclark** [mapnik/node-mapnik#669]
It would be great to have documentation for the Datasource constructor, like the nice docs for other constructors.

**Comment by springmeyer** [comment]
Right, thanks for reminding of this! So, Mapnik's `Datasource` interfaces all accept `key:value` pairs and these are passed to Mapnik core "plugins". Each plugin tends to have custom values it accepts. They are detailed in this JSON doc https://github.com/mapnik/mapnik-reference/blob/gh-pages/3.0.6/datasources.json which is yet to be turned into nice visual documentation. So, the task that needs some help is:

   - Make the datasources show up at http://mapnik.org/mapnik-reference/
   - Do a pass over all the json data to make sure all the `doc` values are filled out
   - Where there are gaps in the `doc` values, ticket missing ones at https://github.com/mapnik/mapnik-reference/issues and recruit @artemp, @GretaCB, or @mapsam to help fill them in by digging through a given plugins code and checking the C++ for the options it accepts: https://github.com/mapnik/mapnik/tree/master/plugins/input
